pkg/configuration/config_manager/handler_util.go

```diff
@@ -189,13 +189,25 @@ func GetSupportReloadConfigSpecs(configSpecs []appsv1alpha1.ComponentConfigSpec,
 	return reloadConfigSpecMeta, nil
 }
 
-func FilterSubPathVolumeMount(metas []ConfigSpecMeta, volumes []corev1.VolumeMount) []ConfigSpecMeta {
+func FilterSupportReloadActionConfigSpecs(metas []ConfigSpecMeta, volumes []corev1.VolumeMount) []ConfigSpecMeta {
```

Comment:
Could add some comments to explain:

// FilterSupportReloadActionConfigSpecs filters the provided ConfigSpecMeta slices based on the reload action type and volume mount configuration.
// It handles two types of updates to ConfigMaps:
//
// 1. Async mode: KubeBlocks controller is responsible for updating the ConfigMap, while kubelet synchronizes the ConfigMap to volumes.
//    The config-manager detects configuration changes using fsnotify and executes the reload action. This requires volume mounting the ConfigMap.
//    However, in async mode, if the volume mount is a subpath, kubelet does not synchronize the ConfigMap content to the container (see https://github.com/kubernetes/kubernetes/issues/50345).
//    As a result, the config-manager cannot detect configuration changes and does not support dynamic parameter updates for such configurations.
//    Therefore, async-type ConfigSpecs with subpath volume mounts need to be removed.
//
// 2. Sync mode: For sync mode (regardless of the reload action type - TPLScriptType trigger or ShellType trigger), the controller directly watches
//    the ConfigMap changes and actively invokes the reload action. 
//
// Both async and sync types need to pass the ConfigSpecs to the config-manager.
//
// The check logic is an OR condition: either it is the first type (sync mode) or the second type (async) with a non-subpath volume mount configuration.

Reply:
updated code comment